generate/templates/client.handlebars

```diff
@@ -140,7 +140,8 @@ func parseAddress(address string) (*url.URL, error) {
 		// used in the application layer and not to the transport layer. Hence,
 		// setting the fields accordingly.
 		parsed.Scheme = "http"
-		parsed.Host = strings.TrimPrefix(address, "unix://") // socket
+		// fake host to construct valid URLs
+		parsed.Host = "unix.socket"
```

Comment:
```suggestion
		parsed.Host = "localhost"
```

Thanks for the contribution, @jakubbujny! Could we make this `localhost` instead to match the [one in vault/api](https://github.com/hashicorp/vault/blob/fd92f2c8ec26f645db810b907f54ed93c004e3f8/api/client.go#L551). It looks like the issue was addressed there a few months ago in https://github.com/hashicorp/vault/pull/22523.